default/python/AI/ShipDesignAI.py

```diff
@@ -1441,6 +1440,66 @@ def _partclass_in_design(self, partclass):
         """
         return any(part.partClass in partclass for part in self.parts)
 
+    def _calculate_weapon_strength(self, weapon_part, ignore_species=False):
+        weapon_name = weapon_part.name
+        base_damage = weapon_part.capacity
+
+        # tech modifiers
+        try:
```

Comment:
May be move such checks to function in `AIDependencies`? 

This is not related to decision making, this is check that developer did not fail.  

IMHO check at module import and fail looks better for me.  This is critical bug and should be found as soon as possible.